diff --git a/base/threading/platform_thread_internal_posix.h b/base/threading/platform_thread_internal_posix.h
--- a/base/threading/platform_thread_internal_posix.h
+++ b/base/threading/platform_thread_internal_posix.h
@@ -41,6 +41,11 @@
 bool SetCurrentThreadTypeForPlatform(ThreadType thread_type,
                                      MessagePumpType pump_type_hint);
 
+#if BUILDFLAG(IS_ALOS)
+bool TrySetCurrentThreadTypeForALOS(ThreadType thread_type,
+                                  MessagePumpType pump_type_hint);
+#endif
+
 #if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)
 // Current thread id is cached in thread local storage for performance reasons.
 // In some rare cases it's important to invalidate that cache explicitly (e.g.
diff --git a/base/threading/platform_thread.cc b/base/threading/platform_thread.cc
--- a/base/threading/platform_thread.cc
+++ b/base/threading/platform_thread.cc
@@ -6,6 +6,9 @@
 
 #include "base/task/current_thread.h"
 #include "base/threading/thread_id_name_manager.h"
+#if BUILDFLAG(IS_ALOS)
+#include "base/threading/platform_thread_internal_posix.h"
+#endif
 #include "base/trace_event/trace_event.h"
 
 #if BUILDFLAG(IS_FUCHSIA)
@@ -64,7 +67,13 @@
 void SetCurrentThreadType(ThreadType thread_type,
                           MessagePumpType pump_type_hint) {
   CHECK_LE(thread_type, ThreadType::kMaxValue);
+#if BUILDFLAG(IS_ALOS)
+  if (!TrySetCurrentThreadTypeForALOS(thread_type, pump_type_hint)) {
+    return;
+  }
+#else
   SetCurrentThreadTypeImpl(thread_type, pump_type_hint);
+#endif
   current_thread_type = thread_type;
 }
 
diff --git a/base/threading/platform_thread_posix.cc b/base/threading/platform_thread_posix.cc
--- a/base/threading/platform_thread_posix.cc
+++ b/base/threading/platform_thread_posix.cc
@@ -123,6 +123,14 @@
                   ThreadType thread_type,
                   MessagePumpType message_pump_type) {
   DCHECK(thread_handle);
+#if BUILDFLAG(IS_ALOS)
+  if (thread_type == ThreadType::kRealtimeAudio) {
+    *thread_handle = PlatformThreadHandle();
+    errno = ENOTSUP;
+    PLOG(ERROR) << "ALOS cannot create a realtime audio thread";
+    return false;
+  }
+#endif
   base::InitThreading();
 
   pthread_attr_t attributes;
@@ -367,6 +375,11 @@
 
 // static
 bool PlatformThreadBase::CanChangeThreadType(ThreadType from, ThreadType to) {
+#if BUILDFLAG(IS_ALOS)
+  (void)from;
+  return to != ThreadType::kRealtimeAudio &&
+         alos_thread_can_set_nice(internal::ThreadTypeToNiceValue(to));
+#else
   if (from >= to) {
     // Decreasing thread priority on POSIX is always allowed.
     return true;
@@ -376,6 +389,7 @@
   }
 
   return internal::CanLowerNiceTo(internal::ThreadTypeToNiceValue(to));
+#endif
 }
 
 namespace internal {
